packages/core/src/delegation/DelegationNode.ts

```diff
@@ -48,18 +61,18 @@ export default class DelegationNode extends DelegationBaseNode
    * @param account Address of the account that will be the owner of the delegation.
    * @param permissions List of [[Permission]]s.
    * @param parentId Identifier of the parent delegation node already stored on-chain. Not required when the parent is the root node.
+   * @param delegationNodeInput
    */
-  public constructor(
-    id: IDelegationNode['id'],
-    rootId: IDelegationNode['rootId'],
-    account: IDelegationNode['account'],
-    permissions: IDelegationNode['permissions'],
-    parentId?: IDelegationNode['parentId']
-  ) {
-    super(id, account)
-    this.permissions = permissions
-    this.rootId = rootId
-    this.parentId = parentId
+  public constructor(delegationNodeInput: IDelegationNode) {
+    super(
+      delegationNodeInput.id,
+      delegationNodeInput.account,
+      delegationNodeInput.revoked
+    )
```

Comment:
100 % on board with changing the constructor interface the way you did, but shouldn't we also change it on the base node as well?

Reply:
Not sure what you mean, since `super(...)` should call the error handling from the base node.

Reply:
That is to say that I don't see a reason why we wouldn't change the `DelegationBaseNode`'s constructor to accept only a single argument as well, which is an object containing an `id`, `account` and `revoked` property.